File: 70-climbing-stairs/70-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        
        int mem[3];
        
        if (n == 1) {
            return 1;
        }
        
        mem[0] = 0;
        mem[1] = 1;
        mem[2] = 2;
        
        for (int i = 3; i <= n; i++){
            int temp = i % 3;
            mem[temp] = mem[(temp + 1) % 3] + mem[(temp + 2) % 3];
        }        
        
        return mem[n % 3];
        
    }
};

File: 70.climbing-stairs.py
# @leet start
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1:
            return 1
        if n == 2:
            return 2
        memo = [0 for _ in range(n)]
        memo[0] = 1
        memo[1] = 2

        for i in range(2, n):
            memo[i] = memo[i-2] + memo[i-1]
        return memo[-1]
# @leet end
